newa/__init__.py

```diff
@@ -968,3 +998,194 @@ def drop_obsoleted_issue(self, issue: Issue, obsoleted_by: Issue) -> None:
                                              transition=self.transitions["dropped"][0])
         except jira.JIRAError as e:
             raise Exception(f"Cannot close issue {issue}!") from e
+
+#
+# ReportPortal communication
+#
+
+
+@define
+class ReportPortal:
+
+    token: str
+    url: str
+    project: str
+
+    def get_launch_url(self, launch_id: str) -> str:
+        return f"{self.url}/ui/#{self.project}/launches/all/{launch_id}"
+
+    def get_request(self,
+                    path: str,
+                    params: Optional[dict[str, str]] = None,
+                    version: int = 1) -> JSON:
+        base_url = f'{self.url}/api/v{version}/{self.project}/{path.lstrip("/")}'
+        get_params = '&'.join([f'{k}={v}' for (k, v) in params.items()]) if params else ''
+        url = f'{base_url}?{get_params}'
```

Comment:
I would strongly recommend using https://docs.python.org/3/library/urllib.parse.html instead of manual processing of URL components. Namely `urlencode` and `urljoin`, take care of the right number of `/` and quoting all parameter names and values. Something like this should give you the right answer:

```python
url = urllib.parse.urljoin(self.url, f'/api/v{version}/{self.project}/{path.lstrip("/")}')

if params:
  url = f'{url}?{urllib.parse.urlencode(params)}'
```

Reply:
Thank you. Fixed.